### PR TITLE
doc: add dataset examples

### PR DESCRIPTION
**Pull request by inashivb**

### CODE REVIEW

File: doc/userguide/rules/datasets.rst [hunk 1]
```diff
@@ -96,6 +96,23 @@ hashsize <size>
 
 .. note:: 'load' and 'state' or 'save' and 'state' cannot be mixed.
 
+Example rules could look like:
+
+1. Detect unique User-Agents::
+
+    alert http any any -> any any (msg:"LOCAL HTTP new UA";http.user_agent;dataset:set,http-ua-seen, type string, state http-ua-seen.csv; sid:8000001; rev:1;)
```
Comment (jufajardini) [inline doc/userguide/rules/datasets.rst:103]: Maybe we could use the rule container? (`.. container:: example-rule`)

Comment (jufajardini) [reply doc/userguide/rules/datasets.rst:103]: (very) nit: space after `msg` semi-colon

File: doc/userguide/rules/datasets.rst [hunk 2]
```diff
@@ -96,6 +96,23 @@ hashsize <size>
...
+
+2. Detect unique TLDs::
+
+    alert dns $HOME_NET any -> any any (msg:"LOCAL DNS unique TLD"; dns.query; pcrexform:"\.([^\.]+)$"; dataset:set,dns-tld-seen, type string, state dns-tld-seen.csv; sid:8000002; rev:1;)
+
+.. image:: dataset-examples/detect-unique-tlds.png
```
Comment (jufajardini) [inline doc/userguide/rules/datasets.rst:109]: While I like using the image to help visualize, I think we're missing either a short intro text to it or a caption, to help guide visualization.

File: doc/userguide/rules/datasets.rst [hunk 3]
```diff
@@ -96,6 +96,23 @@ hashsize <size>
...
+
+Notice how it is not possible to do certain operations alone with datasets
+(example 2 above), but, it is possible to use a combination of other rule
+keywords. Keep in mind the cost of additional keywords though e.g. in the
+rule above, negative performance impact can be expected due to ``pcrexform``.
```
Comment (jufajardini) [inline doc/userguide/rules/datasets.rst:114]:
```suggestion
second example rule above, negative performance impact can be expected due to
``pcrexform``.
```

**Review by jufajardini — CHANGES_REQUESTED** [review]
Just three nit comments, for improving it :)

File: doc/userguide/rules/datasets.rst (hunk 1 above)
Comment (inashivb) [reply doc/userguide/rules/datasets.rst:103]: Didn't know about the rule container. Thanks! :)

Comment (jufajardini) [reply doc/userguide/rules/datasets.rst:103]: There's always something new to learn xD yw!

Comment (inashivb) [reply doc/userguide/rules/datasets.rst:103]: @jufajardini sorry for coming back late at this but do you want space after every semi-colon separated field or just the `msg`?

Comment (jufajardini) [reply doc/userguide/rules/datasets.rst:103]: Good point! Whenever it happens (except for the parenthesis case :P)
It increases readability for human folks, imho.

Comment (inashivb) [reply doc/userguide/rules/datasets.rst:103]: nvm there aren't any more

**Pull request closed without merging**